packages/itwinui-css/src/surface/surface.scss

```diff
@@ -22,5 +22,43 @@
 
 // Public facing surface component.
 @mixin iui-surface-component {
+  --iui-surface-padding-inline: var(--iui-size-s);
+  --iui-surface-body-padding: var(--iui-surface-padding-inline);
+
   @include iui-surface;
+  display: flex;
+  flex-direction: column;
+  height: var(--iui-surface-height);
+}
+
+@mixin iui-surface-header-footer($type) {
+  display: flex;
+  flex-shrink: 0;
+  align-items: center;
+  overflow: hidden;
+
+  @if ($type == 'header') {
+    border-bottom: 1px solid var(--iui-surface-border-color);
+    padding: calc(var(--iui-surface-padding-inline) * 0.5) var(--iui-surface-padding-inline);
+  }
+
+  @if ($type == 'footer') {
+    border-top: 1px solid var(--iui-surface-border-color);
+  }
+}
+
+@mixin iui-surface-header {
+  @include iui-surface-header-footer(header);
+}
+
+@mixin iui-surface-footer {
+  @include iui-surface-header-footer(footer);
+}
+
+@mixin iui-surface-body {
+  display: flex;
+  flex-grow: 1;
+  padding: var(--iui-surface-body-padding);
+  overflow-y: auto;
+  overflow-y: overlay;
```

Comment:
Might want to swap out `overflow: overlay` with the newer [`scrollbar-gutter: stable`](https://developer.mozilla.org/en-US/docs/Web/CSS/scrollbar-gutter) which works across all browsers.

Reply:
`stable both-edges` would also be a very nice touch for the padded variant.

Reply:
Thanks for sharing this, I love learning about new CSS props like `scrollbar-gutter`!

Right now [our scrollbar width is set to `var(--iui-size-xs)`](https://github.com/iTwin/iTwinUI/blob/a0693430dba7c9e9e52f2b0ca252bdee43fdfbff/packages/itwinui-css/src/utils/mixins.scss#L41) but we've been using `var(--iui-size-s)` for our padded areas.

What I'd like to propose: 1 PR that applies `scrollbar-gutter` everywhere applicable & change the scrollbar to [this old branch](https://github.com/iTwin/iTwinUI/blob/829e18ccb83a04ac799da3040ff7c834c42aa5f6/packages/itwinui-css/src/utils/mixins.scss#L36) that indents the scrollbar from the edge of the area. This would result in the same padding being applied currently. (Image below from a concept image on old documentation site). 
![image](https://user-images.githubusercontent.com/849817/219428453-eb86ae93-fbdb-45d5-8c47-4b72c387889e.png)

Reply:
the padding is currently messed up in the changes you pushed.

what i'm suggesting: use `stable both-edges` *along with* a smaller padding.

Reply:
I believe this is fixed now.